source/common/router/router.cc

```diff
@@ -222,6 +222,29 @@ Filter::~Filter() {
   ASSERT(!retry_state_);
 }
 
+const FilterUtility::StrictHeaderChecker::HeaderCheckResult
+FilterUtility::StrictHeaderChecker::test(Http::HeaderMap& headers,
+                                         const Http::LowerCaseString& target_header) {
+  HeaderCheckResult r;
```

Comment:
This doesn't seem to be used besides returning? Might be easier to just inline it e.g. `return HeaderCheckResult();`

Reply:
Good catch; this is was from a time when we always returned "valid" to ignore a `target_header` didn't have an associated validator. This in combination with the assert + `NOT_REACHED_GCOVR_EXCL_LINE` below means we ought to be able to remove both this declaration and its return statement.